### ISSUE TITLE
"Detect Conflict At Sync" doesn't work

### ISSUE THREAD

**Issue by darioms21** [forcedotcom/salesforcedx-vscode#3522]
### Summary

Click on "SFDX: Deploy Source in Manifest to Org" on Packaged ("salesforcedx-vscode-core.detectConflictsAtSync": true); The conflict start and run success aleatory.

### Steps To Reproduce:

1. Enable "Salesforcedx-vscode-core: Detect Conflicts At Sync".
2. Change metadata.
3. Click "SFDX: Deploy Source in Manifest to Org" on Packaged.
4. Running "SFDX: Deploy Source in Manifest to Org" does not always fail.

### Expected result

Always that Enable "Salesforcedx-vscode-core: Detect Conflicts At Sync". Should run the conflict and when there are conflicts show a alert to view the conflict or override the source.

### Actual result

The confict never correctly executed. Only occasionally in action Deploy but not always.

### Additional information
![image](https://user-images.githubusercontent.com/75618319/131335637-b1f0f217-542d-43c3-9e6b-e16766c8631b.png)
![image](https://user-images.githubusercontent.com/75618319/131335713-1f674e91-5019-4a17-97e6-f80dfe85e6f9.png)
![image](https://user-images.githubusercontent.com/75618319/131335762-a610c9b7-9d3b-404a-b66b-0e6c20819890.png)
![image](https://user-images.githubusercontent.com/75618319/131335842-3ef49ea0-51d5-406e-8484-1c95a5e2a6d5.png)
The Conflict Detection doesn't detect my change in the title of the component for example.

**Salesforce Extension Version in VS Code**:
v52.8.0

**SFDX CLI Version**:
sfdx-cli/7.110.0 win32-x64 node-v14.17.3

**OS and version**:
OS: Windows 10 Enterprise
Version: 1909

**Comment by PawelWozniak** [comment]
I also experienced issues with conflict detection. First I have pushed metadata from the local computer to scratch org then on each save of class I was notified that there is a conflict with the version on org. But no changes were done there, I am the only user o this scratch org.
Also when I selected to overwrite changes on org the next save also displayed a notification about conflict. 
Finally, I deactivated that because is getting many false positive alarms.
Diff file against org shows that there were no changes on org.

**Comment by PawelWozniak** [comment]
It might be related to this isue https://github.com/forcedotcom/cli/issues/1178 .
It seems to be similar that force:source:status show files that were just saved to SF as conflict and conflict detection is also alarming about those files. 

The main issue is that file that was just saved to SF somehow is modified on SF servers and appears as changed. 
However, as described in the other bug report pulling those files to the local file system doesn't indicate any changes.

**Comment by ralphcallaway** [comment]
@PawelWozniak the order that stuff happens
- file saved locally
- deployment!!
- file saved remotely

Except in the best cases the time it takes to actually deploy means the local file modified date will always be before the remove file modified date, even though they were changed at the same time. Wooops! 

See #3542

**Comment by PawelWozniak** [comment]
@ralphcallaway Good that we know at least what is the source of the issue. Maybe it could be replaced by generating a hash of the file on the local and server-side and comparing those hashes.

**Comment by floralan** [comment]
@darioms21 This issue has been resolved here [#3556](https://github.com/forcedotcom/salesforcedx-vscode/pull/3556). Could you verify it has been fixed in our latest release?

**Comment by ogomezba** [comment]
> @darioms21 This issue has been resolved here [#3556](https://github.com/forcedotcom/salesforcedx-vscode/pull/3556). Could you verify it has been fixed in our latest release?

Hi @floralan,

I've done some checks and I think that the issue reported in #3542 (which was closed in favour of this issue) is still there. I'm not sure if I need to do any additional steps or enable something but if I follow the steps mentioned in #3542, I still reproduce the same error regarding conflict detection (as explained by @ralphcallaway):

> ### Steps To Reproduce:
> Open SomeClass.cls
> Run the "SFDX: Retrieve this Source from Org" command
> Make a change to SomeClass.cls and save
> Make a second change and save
> 
> ### Expected result
> Both saves should go through since the version of the class prior to the change is up to date.
> 
> ### Actual result
> The first change after doing a retrieve goes through without throwing a conflict error. The second change throws a conflict error.

**Comment by PawelWozniak** [comment]
I can confirm this. 
I have:
1. Updated SFDX to sfdx-cli/7.144.2 win32-x64 node-v16.14.2 
2. Reset source tracking with force:source:tracking:reset (and legacy reset as format of source tracking is changed as described [here](https://github.com/forcedotcom/cli/blob/main/releasenotes/sfdx/README.md#71400-march-3-2022)) 
3. Enabled "Detect conflict at sync" after updating sfdx

And still on every save of file I am getting warning:
<img width="411" alt="2022-04-07_09h20_19" src="https://user-images.githubusercontent.com/5740564/162142605-f3d02673-d92a-4fce-9bd5-fc40f411440b.png">

Reproduce:
1. Make changes in code.
4. Click override to force save your changes. You expect at this moment that files have the same version.
5. Add a line of code and save.
6. You get the same popup that file have conflicts.
7. Click "View Conflicts and Cancel deploy"
8. See that new line is a conflict.

<img width="821" alt="2022-04-07_09h23_58" src="https://user-images.githubusercontent.com/5740564/162144078-23febf29-533e-42d4-988c-24b36be4620f.png">




Also after clicking override whole saving process is significantly longer than without "Detect conflict at sync" active.
 
sfdx-cli/7.144.2 win32-x64 node-v16.14.2 
VSCode: 1.66, 
SF plugin: v54.7.0

**Comment by ralphcallaway** [comment]
please reopen

this is a bit frustrating since you a) have multiple working examples of this (force.com ide, mavensmate) the later is open source and b) there have been repeated explanations that you can't use the "local" file save time as the last modified time, you need to use the "remote" file completed saving and mark that as the "local" file save time (either by touching the file or retrieving it, or getting into a basic db tracking).

You have working prototypes in mavensmates, please make sure you're devs review this. Right now I have my whole team turning this off which sucks because people are overriding each others stuff and it's caused multiple deployment issues (woops, the apex in my vs code isn't wants in the change set, huh???). 

It's been years you've been working on a very basic feature with tons of community support the whole time. Frankly it's embarassing for Salesforce. I know this is kind of "open source", but I don't think you get that excuse when it's a tool for a commercial product. You've gone through three iterations so far. Please, please, please, please 1) prioritize fixing this!!!! and 2) fix it properly. Feel free to reach out to me if you need guidance.

**Comment by ralphcallaway** [comment]
@floralan plesea re-open so this can get visibility

**Comment by ralphcallaway** [comment]
@smaddox-sf can you assist in re-opening this perennial misbehaving feature

**Comment by AnanyaJha** [comment]
Hi @ralphcallaway checking in to see if you're still experiencing this issue?

**Comment by ogomezba** [comment]
Hi @AnanyaJha,

I just checked a few minutes ago and the issue is still there. Once the option is enabled (Detect Conflicts At Sync), every time I try to deploy a new version of the file it is always detected as a conflict.

**Comment by AnanyaJha** [comment]
Hi @ogomezba ! Thanks for reaching out - could you open up a new issue with more details on the problem you're facing and what version of the Salesforce Extensions you're using? We'll have the team take a look at your issue, thank you!

**Comment by ralphcallaway** [comment]
@AnanyaJha i've got copious details already available in #3542

i really appreciate the follow up. this has felt a bit ridiculous. I worked on this exact same feature with maven's mate (precursor to vs code extension), and documented how to do this correctly in #2034. when it wasn't done correctly i documented it again in #3542. each time the tickets get closed without anything actually getting fixed. it's painful given the exact solution is already outlined for engineering. this feature has been released twice now over two years, and still remains completely unusable :(
